app/api/server/v1/rooms.js

```diff
@@ -218,3 +218,54 @@ API.v1.addRoute('rooms.leave', { authRequired: true }, {
 		return API.v1.success();
 	},
 });
+
+API.v1.addRoute('rooms.createDiscussion', { authRequired: true }, {
+	post() {
+		const { parentRoomId, parentMessageId, reply, name, users } = this.bodyParams;
+		if (!parentRoomId) {
+			return API.v1.failure('Body parameter "parentRoomId" is required.');
+		}
+		if (!name) {
+			return API.v1.failure('Body parameter "name" is required.');
+		}
+		if (users && !Array.isArray(users)) {
+			return API.v1.failure('Body parameter "users" must be an array.');
+		}
+
+		const discussion = Meteor.runAsUser(this.userId, () => Meteor.call('createDiscussion', {
+			prid: parentRoomId,
+			pmid: parentMessageId,
+			t_name: name,
```

Comment:
@ggazzo why we are using `t_name` here?

Reply:
@rodrigok I dont get it, `t_name` is the `fname`, you can choose any name you want... 

I'm not sure if we should use `parentRoomId` instead of `prid` because the methods will return the short form...

Reply:
please don't rename parameters.

Reply:
I agree, but `t_name` is outside of camel case convention.

Reply:
@ggazzo We can't choose any name since the method uses named parameters https://github.com/RocketChat/Rocket.Chat/blob/936ced9991ad98dcbb8c04330dce9cdeb3a63689/app/discussion/server/methods/createDiscussion.js#L116